File: Lab6/C++/ConsoleApplication1/ConsoleApplication1/TextContainer.cpp
#include <vector>
#include "TextContainer.h"
#include "MyString.h"

void TextContainer::addLine(MyString& line) {
	data.push_back(line);
}

void TextContainer::deleteLine(size_t index) {
	data.erase(data.begin() + index);
}

void TextContainer::clearText() {
	data.clear();
}

size_t TextContainer::totalChars() {
	size_t total = 0;

	for (auto& line : data) {
		total += line.length();
	}

	return total;
}

int TextContainer::findLine(std::string lineToFind) {
	int index = 0;

	for (auto& line : data) {
		if (line.get() == lineToFind) {
			break;
		}
		index++;
	}

	return index;
}

void TextContainer::replaceCharacter(char newCh, char oldCh) {
	for (auto& line : data) {
		line.replaceChar(newCh, oldCh);
	}
}

File: Lab6/C#/Container/Container/TextContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Container
{
    internal class TextContainer
    {
        private List<MyString> text;

        public void addLine(MyString line)
        {
            text.Add(line);
        }

        public void deleteLine(int index)
        {
            text.RemoveAt(index);
        }

        public void clearText()
        {
            text.Clear();
        }

        public int totalChars()
        {
            int total = 0;
            foreach (MyString s in text)
            {
                total += s.Length();
            }

            return total;
        }

        public int findLine(string line)
        {
            int index = 0;

            foreach (MyString s in text)
            {
                if(s.getData() == line)
                {
                    break;
                }

                index++;
            }

            return index;
        }

        public void replaceCharacher(char oldCh, char newCh)
        {
            foreach (MyString s in text)
            {
                s.replaceChar(oldCh, newCh);
            }
        }
    }

}
